controllers/clusterversion_controller.go

```diff
@@ -58,7 +60,28 @@ func (r *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	if err := r.Client.Get(context.TODO(), req.NamespacedName, &instance); err != nil {
 		return ctrl.Result{}, err
 	}
-	if err := r.ensureConsolePlugin(instance.Status.Desired.Version); err != nil {
+
+	if err := r.ensureOdfConsoleConfigMapAndService(); err != nil {
+		logger.Error(err, "Could not ensure configmap and service for odf-console deployment")
+		return ctrl.Result{}, err
+	}
+
+	csvList, err := util.GetNamespaceCSVs(ctx, r.Client, r.OperatorNamespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Check if it is upgrade from 4.17 to 4.18
+	// The new CSVs won't exists while upgrading
+	// They will exists only after new operator has created a new subscription
+	if !util.AreMultipleOdfOperatorCsvsPresent(csvList) {
```

Comment:
just a general comment/doubt (no need to change the current implementation):

this controller will only run when either OCP is upgraded (i.e., For(&configv1.ClusterVersion{})) or when controller will restart after ODF is upgraded (in which case new CSVs will be existing soon)... we don't really need `!util.AreMultipleOdfOperatorCsvsPresent(csvList)` check, right ?? or am I missing any case ??

Reply:
This check is to make sure that we don't block the reconcile in case of upgrades. For example assume customers has messed up the catalog for some reason in that case only odf will be upgraded and other operators wont be upgraded. So we should not block the console updation in that case right?

Reply:
> So we should not block the console updation in that case right?

make sense.

> For example assume customers has messed up the catalog for some reason in that case only odf will be upgraded and other operators wont be upgraded.

If ODF is upgraded and others won't, then we are still blocked right ?? because `!util.AreMultipleOdfOperatorCsvsPresent(csvList)` will be `true` in that case.

Reply:
Yes, that's right, It will skip if the odf-operator upgrade is in progress.

Reply:
actually that was my initial doubt, why would this code be executed when ODF upgrade is in progress: https://github.com/red-hat-storage/odf-operator/pull/511#discussion_r1875486724

Reply:
Once the manager is started, The controllers will also get triggered. 

I have added this check to keep the behavior as before and that is once the odf-operator is upgraded, I also upgrade the console plugin.